x-pack/test/fleet_api_integration/apis/agent_policy/agent_policy.ts

```diff
@@ -437,6 +438,86 @@ export default function (providerContext: FtrProviderContext) {
 
         expect(policy.package_policies[0].name).be('system-457');
       });
+
+      it('should create policy with global data tags given valid tags', async () => {
+        const {
+          body: { item: createdPolicy },
+        } = await supertest
+          .post(`/api/fleet/agent_policies?sys_monitoring=true`)
+          .set('kbn-xsrf', 'xxxx')
+          .send({
+            name: 'global data tag test',
+            namespace: 'default',
+            global_data_tags: [
+              { name: 'testName', value: 'testValue' },
+              { name: 'testName2', value: 123 },
+            ],
+          })
+          .expect(200);
+
+        let res = await supertest.get(`/api/fleet/agent_policies/${createdPolicy.id}`).expect(200);
+        jestExpect(res.body.item.global_data_tags).toEqual([
+          { name: 'testName', value: 'testValue' },
+          { name: 'testName2', value: 123 },
+        ]);
+
+        res = await supertest.get(`/api/fleet/agent_policies/${createdPolicy.id}/full`).expect(200);
+        for (const input of res.body.item.inputs) {
+          jestExpect(input.processors).not.toBeUndefined();
+          jestExpect(input.processors.length).toEqual(1);
+          const addFields = input.processors[0].add_fields;
+          jestExpect(addFields).toEqual({
+            fields: { testName: 'testValue', testName2: 123 },
+            target: '',
+          });
+        }
+      });
+
+      it('should not create agent policy if provided with global data tag values that are not strings or numbers', async () => {
```

Comment:
I think the validation tests would be more efficient as unit tests.

Reply:
I removed all validation tests, kept the happy path tests.